### PR TITLE
Updated for Unicorn rules

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

-   [ ] Fix
-   [ ] Improvements
-   [ ] New Feature
-   [ ] Documentation updates
-   [X] Other, please explain:

## What changes did you make? (Give an overview)

added unicorn for eslint based improvements

...

### CODE REVIEW

**Comment by JoernBerkefeld** [comment]
Merge conflicts @DougMidgley ?

**Comment by DougMidgley** [comment]
@JoernBerkefeld  - sorry not sure how they got there :) resolved now

File: lib/auth.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (JoernBerkefeld) [inline lib/auth.js:1]: why remove that?

File: lib/rest.js [hunk 2]
```diff
@@ -90,7 +88,7 @@ module.exports = class Rest {
                 page++;
                 shouldPaginate = true;
                 if (this.options?.eventHandlers?.onLoop) {
-                    this.options.eventHandlers.onLoop(null, collector?.[iteratorField]);
+                    this.options.eventHandlers.onLoop(undefined, collector?.[iteratorField]);
```
Comment (JoernBerkefeld) [inline lib/rest.js:91]: why? that is a potentially breaking change for users of the SDK

File: lib/rest.js [hunk 3]
```diff
@@ -52,36 +50,36 @@ module.exports = class Rest {
...
+        const queryParameters = new URLSearchParams(url.split('?')[1]);
         let collector;
         let shouldPaginate = false;
-        queryParams.set('$pageSize', Number(pageSize || 50).toString());
+        queryParameters.set('$pageSize', Number(pageSize || 50).toString());
         do {
-            queryParams.set('$page', Number(page).toString());
-            const temp = await this._apiRequest(
+            queryParameters.set('$page', Number(page).toString());
+            const temporary = await this._apiRequest(
```
Comment (JoernBerkefeld) [inline lib/rest.js:59]: if we are renaming it - wouldn't `reponseBatch` or sth similar be more fitting?

File: lib/soap.js [hunk 4]
```diff
@@ -440,18 +444,18 @@ module.exports = class Soap {
...
                 return this._apiRequest(options, 1);
-            } else if (ex instanceof SOAPError) {
+            } else if (error instanceof SOAPError) {
                 //rethrow as is already handled/parsed
-                throw ex;
+                throw error;
             } else {
                 //unknown error
-                throw new SOAPError(ex, response, null);
+                throw new SOAPError(error, response, undefined);
```
Comment (JoernBerkefeld) [inline lib/soap.js:458]: checked what SOAPError is expecting and testing for in it's 3rd parameter?

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

File: lib/rest.js [hunk 5]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply lib/rest.js:1]: not sure, reverted

File: package.json [hunk 6]
```diff
@@ -1,50 +1,52 @@
...
-    "nyc": "15.1.0",
-    "prettier-eslint": "15.0.1",
-    "sinon": "15.0.1"
-  },
-  "engines": {
-    "npm": ">=6.14.4",
-    "node": ">=14.0.0"
-  }
+    "name": "sfmc-sdk",
+    "version": "0.6.3",
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply package.json:3]: good catch, reverted

File: lib/rest.js (hunk 3 above)
Comment (DougMidgley) [reply lib/rest.js:59]: good idea :)

File: lib/rest.js [hunk 7]
```diff
@@ -52,36 +50,36 @@ module.exports = class Rest {
...
-                throw new Error('Could not find an array to iterate over');
+                throw new TypeError('Could not find an array to iterate over');
             }
-            if (collector && Array.isArray(temp[iteratorField])) {
-                collector[iteratorField].push(...temp[iteratorField]);
-            } else if (collector == null) {
-                collector = temp;
+            if (collector && Array.isArray(temporary[iteratorField])) {
+                collector[iteratorField].push(...temporary[iteratorField]);
+            } else if (collector == undefined) {
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply lib/rest.js:75]: would be safe :) but indeed moved to !collector.

File: lib/rest.js (hunk 2 above)
Comment (DougMidgley) [reply lib/rest.js:91]: good point, put a todo to replace when moving to v1 ( potential breaking change)

File: lib/soap.js (hunk 4 above)
Comment (DougMidgley) [reply lib/soap.js:458]: it is a safe change for the error, realistically, i would be happy to take risk here on error handling, but if you are concerned, can also move to v1 change

Comment (JoernBerkefeld) [reply lib/soap.js:458]: bad wording on my part: "did YOU check..."? :D if things worked fine in your tests then I'm good. Though I gotta admit I don't understand why null has to become undefined. We've been using null across projects for a while now in these cases

Comment (JoernBerkefeld) [reply lib/soap.js:458]: did you? @DougMidgley

Comment (DougMidgley) [reply lib/soap.js:458]: Yes tested while running the rate limiting tests without issue.

**Pull request merged**